code/src/components/InputField.js

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import "./InputField.css";
+
+const InputField = ({ name, type, options, value, onChange }) => {
+  switch (type) {
+    case "text": {
+      return (
+        <input
+          id={name}
+          value={value}
+          onChange={onChange}
+          type={type}
+          placeholder="Type here..."
+          autoFocus
+          required
+        ></input>
+      );
+    }
+    case "radio": {
+      return options.map((o) => {
+        return (
+          <div key={o.value}>
+            <input
+              id={o.value}
+              type={type}
+              name={name}
+              value={o.value}
+              checked={value === o.value}
+              onChange={onChange}
+              required
+            />
+            <label htmlFor={o.value}>{o.label}</label>
+          </div>
+        );
+      });
+    }
+
+    case "select": {
+      return (
+        <select value={value} onChange={onChange} required>
+          <option value="" defaultValue disabled>
+            Choose option
+          </option>
+          {options.map((o) => (
+            <option key={o.value} value={o.value}>
+              {o.label}
+            </option>
+          ))}
+        </select>
+      );
+    }
+    case "range": {
+      return (
+        <>
+          <label htmlFor={name}>
+            <input
+              type={type}
+              id={name}
+              min={options[0].value}
+              max={options[options.length - 1].value}
+              defaultValue={options[0].value}
+              onChange={onChange}
+            />
+          </label>
+          <div className="input-range-group">
+            {options.map((o) => (
+              <span key={o.value}>{o.label}</span>
+            ))}
+          </div>
+        </>
+      );
+    }
+    default:
+      console.log("Error: type can only be 'text', 'radio', 'range', or 'select'");
```

Comment:
Well done catching this state! Preferably we will never get here, but in case we do - what is presented to the user? Maybe we should show them some error message and have room for logging.

Reply:
Smart! Related - showing an error message is a stretch goal for this week's project that I want to try out. I'll think about how to go about this. Good feedback!